### ISSUE TITLE
As of 0.8.3 webpackMerge mutates the inputs

### ISSUE THREAD

**Issue by sullivanpt** [survivejs/webpack-merge#12]
This is a new behavior that didn't occur in 0.7.x; I'm not sure its a "bug" but it should at least be documented.  The webpackMerge function mutates the inputs, as evidenced by the unit test below.  It's a problem because I've been using webpackMerge in series to build multiple distinct configurations, for example in webpack.config.js:

```
module.exports = [
  webpackMerge({}, defaultConfig, commonConfig, clientConfig),
  webpackMerge({}, defaultConfig, commonConfig, serverConfig)
]
```

The failing test:
```
  it('should not mutate inputs', function () {
    const a = {
      output: {
        filename: 'bundle.js'
      }
    };
    const b = {
      output: {
        path: 'path/b'
      }
    };

    const aClone = JSON.parse(JSON.stringify(a));
    merge({}, a, b);

    assert.deepEqual(a, aClone); // this fails with actual: -    "path": "path/b"
  });
```

**Comment by gdi2290** [comment]
I use webpack-merge everywhere now but this is becoming a problem for universal applications

**Comment by bebraw** [comment]
Thanks for reporting! I was missing a little corner case there. Fixed in `0.8.4`.

**Comment by gdi2290** [comment]
thanks!
